block_alerts.php

```diff
@@ -77,19 +77,46 @@ public function get_content(): stdClass {
         return $this->content;
     }
 
+    /**
+     * Return if user has archetype editingteacher.
+     *
+     */
+    public static function is_teacher(): bool {
+        global $DB, $USER;
+        // Get id's from role where archetype is editingteacher.
+        $roles = $DB->get_fieldset('role', 'id', ['archetype' => 'editingteacher']);
+
+        // Check if user has editingteacher role on any courses.
+        list($roles, $params) = $DB->get_in_or_equal($roles, SQL_PARAMS_NAMED);
+        $params['userid'] = $USER->id;
+        $sql = "SELECT id
+                FROM {role_assignments}
+                WHERE userid = :userid
+                AND roleid $roles";
+        return  $DB->record_exists_sql($sql, $params);
+    }
+
     /**
      *  Get the alerts.
      *
      * @return array alerts items.
      */
     public function fetch_alert(): array {
+        // Staff only check.
+        if (get_config('block_alerts', 'staffonly')) {
```

Comment:
Could be simplify like:
```
if (get_config('block_alerts', 'staffonly') && !self::is_teacher()) {
    return []; // Don't output for learners.
}
```

Reply:
@aydevworks 👍 

The current check does a none expensive call - get_config('block_alerts', 'staffonly')

If that evaluates as true we call the expensive self::is_teacher();

Would  if (get_config('block_alerts', 'staffonly') && !self::is_teacher())  have the same performance or would it always calls self::is_teacher?

Reply:
If get_config('block_alerts', 'staffonly') is false: The second condition (!self::is_teacher()) will not be executed.